doc/design/context.md

```diff
@@ -0,0 +1,166 @@
+## Context Design
+
+
+A Net is executed by single or several threads. A Context is related to a thread and records necessary runtime resources.
+
+Context is defined as follows:
+
+```
+class Context {};
+```
+
+### CUDAContext
+
+Because the Tensor computation are executed by Eigen library, which needs an Eigen::GpuDevice type object as  parameter. And the GpuDevice parameter is constructed with an Eigen::CudaStreamDevice object. We need to set a specific GpuID and CudaStream to create a Eigen::CudaStream object.
+
+At the same time, some computation work will executed by cublas or cudnn library. Take cublas library as an example, we have to acquire a cublasHandle which binds on a CudaStream to make computation. It's the same way as Eigen library does.
+
+The future DAGNet is run by multi-threads. And each thread will have its own Eigen::GpuDevice object binding on different CudaStream. Multi-threads can run parallelly on a same GPU card.
```

Comment:
all ". And" -> ", and"

on different CudaStream. Multi-threads can run parallelly on a same GPU card. 
-> 
on different CudaStream**, so that** multi-threads can run parallelly on **the** same GPU card.